drivers/include/periph/spi.h

```diff
@@ -1,192 +1,174 @@
+} spi_mode_t;
+#endif
 
 /**
- * @brief Initialize the given SPI device to work in slave mode
+ * @brief   Available SPI clock speeds
  *
- * In slave mode the SPI device is purely reacting to the bus. Transaction will be started and
- * ended by a connected SPI master. When a byte is received, the callback is called in interrupt
- * context with this byte as argument. The return byte of the callback is transferred to the
- * master in the next transmission cycle. This interface enables easy implementation of a register
- * based access paradigm for the SPI slave.
- *
- * @param[in] dev       The SPI device to initialize as SPI slave
- * @param[in] conf      Mode of clock phase and polarity
- * @param[in] cb        callback called every time a byte was received
- *
- * @return              0 on success
- * @return              -1 on error
+ * The actual speed of the bus can vary to some extend, as the combination of
+ * CPU clock and available prescaler values on certain platforms may not make
+ * the exact values possible.
  */
-int spi_init_slave(spi_t dev, spi_conf_t conf, char (*cb)(char data));
+#ifndef HAVE_SPI_CLK_T
+typedef enum {
+    SPI_CLK_100KHZ,         /**< drive the SPI bus with 100KHz */
+    SPI_CLK_400KHZ,         /**< drive the SPI bus with 400KHz */
+    SPI_CLK_1MHZ,           /**< drive the SPI bus with 1MHz */
+    SPI_CLK_5MHZ,           /**< drive the SPI bus with 5MHz */
+    SPI_CLK_10MHZ           /**< drive the SPI bus with 10MHz */
+} spi_clk_t;
+#endif
 
 /**
- * @brief Configure SCK, MISO and MOSI pins for the given SPI device
+ * @brief   Initialize pins used by the given SPI device
  *
- * @param[in] dev       SPI device to use
+ * @param[in] dev       initialize pins for this SPI device
+ * @param[in] cs        also initialize this chip select pin
  *
  * @return              0 on success
- * @return              -1 on error
+ * @return              -1 on invalid device
+ * @return              -2 on invalid CS
  */
-int spi_conf_pins(spi_t dev);
+int spi_init(spi_t dev, spi_cs_t cs);
 
 /**
- * @brief Get mutually exclusive access to the given SPI bus
+ * @brief   Start a new SPI transaction
  *
- * In case the SPI device is busy, this function will block until the bus is free again.
+ * Starting a new SPI transaction will get exclusive access to the SPI bus
+ * and configure it according to the given values. If another SPI transaction
+ * is active when this function is called, this function will block until the
+ * other transaction is complete (spi_relase was called).
  *
  * @param[in] dev       SPI device to access
+ * @param[in] mode      mode to use for the new transaction
+ * @param[in] clk       bus clock speed to use for the transaction
+ * @param[in] cs        chip select pin to use
  *
  * @return              0 on success
  * @return              -1 on error
  */
-int spi_acquire(spi_t dev);
+int spi_acquire(spi_t dev, spi_mode_t mode, spi_clk_t clk, spi_cs_t cs);
```

Comment:
I don't agree with having CS as both part of the acquire call and the `spi_transfer_x` calls.
The API would be slimmer if either the acquire call stores the CS value for the transaction, or if the acquire call doesn't care about CS and let each transfer function take it as an argument, instead of both.